### PR TITLE
Add in the miq. automate and vmdb roles

### PR DESCRIPTION
**Pull request by syncrou**
The roles have been namespaced via the `miq` prefix: `miq.manageiq-automate` and `miq.manageiq-vmdb`.

https://bugzilla.redhat.com/show_bug.cgi?id=1539762

### CODE REVIEW

**Comment by syncrou** [comment]
@miq-bot add_label wip, enhancement

@miq-bot assign @Fryguy 

cc - @gmcculloug, @gtanzillo

**Comment by syncrou** [comment]
@miq-bot remove_label wip

**Comment by gmcculloug** [comment]
@Fryguy I think this is ready to merge.  Any remaining issues?

**Comment by JPrause** [comment]
@Fryguy it appears this can be merged. Can you review.

**Comment by jrafanie** [comment]
@gmcculloug @syncrou can you drop the WIP from the description of this PR if it's ready?

**Comment by syncrou** [comment]
@jrafanie - I pulled out the WIP statement.

**Comment by jrafanie** [comment]
@miq-bot add_label blocker

File: content/ansible/roles/miq-core.manageiq-automate/tests/test.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+- hosts: localhost
+  remote_user: root
+  roles:
+    - ansible_manageiq
```
Comment (Fryguy) [inline content/ansible/roles/miq-core.manageiq-automate/tests/test.yml]: Is this role correct?  I thought it was manageiq-automate

File: content/ansible/roles/miq-core.manageiq-vmdb/tests/test.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+- hosts: localhost
+  remote_user: root
+  roles:
+    - ansible_manageiq
```
Comment (Fryguy) [inline content/ansible/roles/miq-core.manageiq-vmdb/tests/test.yml]: Same here.

**Comment by Fryguy** [comment]
Any way we can make the prefix manageiq-core instead of miq-core?

**Comment by syncrou** [comment]
@Fryguy - renamed all references to `miq-core` to `manageiq-core`

File: content/ansible/roles/manageiq-core.manageiq-automate/meta/main.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+   - name: Fedora
+     versions:
+      - 25
+      - 26
+   - name: Ubuntu
+     versions:
+      - xenial
+      - precise
+  galaxy_tags: [manageiq, cloudforms, automate, rhel]
```
Comment (Fryguy) [inline content/ansible/roles/manageiq-core.manageiq-automate/meta/main.yml]: @syncrou Why have the galaxy stuff in here?  If we keep it, I think we should

- remove cloudforms tag
- change author to "ManageIQ Developers" as we do for the [rest of the manageiq-content](https://github.com/ManageIQ/manageiq-content/blob/master/manageiq-content.gemspec#L10).

Are the platforms necessary?  I'd hate to have to maintain that list.

File: content/ansible/roles/manageiq-core.manageiq-vmdb/README.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
+manageiq-core.manageiq-vmdb
+=========
+
+The `manageiq-core.manageiq-vmdb` role allows for users of ManageIQ to modify and/or change VMDB objects via an Ansible Playbook.
+The role includes a module `manageiq_vmdb` which does all the heavy lifting needed to modify or change objects in the database.
+
+Requirements
+------------
+
+ManageIQ has to be Gaprindashvili (G Release) or higher.
```
Comment (Fryguy) [inline content/ansible/roles/manageiq-core.manageiq-vmdb/README.md]: Minor, but the REST API actually has a separate versioning scheme, so it's probably better to align to the API version.

File: content/ansible/roles/manageiq-core.manageiq-vmdb/README.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+ManageIQ:
+    `manageiq_connection` is a dictionary with connection default keys.
+    Use of this connection information is ONLY needed if the role is used outside of a ManageIQ
+    appliance. A ManageIQ appliance passes in `manageiq_connection` via `extra_vars` so connection
+    information is included automatically.
+    Remember to use Ansible Vault for passwords.
+
+```
+    manageiq_connection:
+        url: 'http://localhost:3000'
```
Comment (Fryguy) [inline content/ansible/roles/manageiq-core.manageiq-vmdb/README.md]: probably should be https.

File: content/ansible/roles/manageiq-core.manageiq-vmdb/library/manageiq_vmdb.py [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+
+    def build_url(self):
+        """
+            Using any type of href input, build out the correct url
+        """
+
+        url_actual = urlparse(self._href)
+        if re.search('api', url_actual.path):
+            return self._api_url + url_actual.path
+        return self._api_url + '/api/' + url_actual.path
```
Comment (Fryguy) [inline content/ansible/roles/manageiq-core.manageiq-vmdb/library/manageiq_vmdb.py]: Minor but why have a separate url and build_url method?  I feel like the content of build_url could just be in url itself.

File: content/ansible/roles/manageiq-core.manageiq-vmdb/meta/main.yml [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+   - name: Fedora
+     versions:
+      - 25
+      - 26
+   - name: Ubuntu
+     versions:
+      - xenial
+      - precise
+  galaxy_tags: [manageiq, cloudforms, automate, rhel]
```
Comment (Fryguy) [inline content/ansible/roles/manageiq-core.manageiq-vmdb/meta/main.yml]: Same here.

**Comment by gmcculloug** [comment]
@Fryguy Can you point to an example of what is required for yamllint?  Depending on the level of effort it could be a follow PR.

File: content/ansible/roles/manageiq-core.manageiq-vmdb/README.md (hunk 4 above)
Comment (syncrou) [reply content/ansible/roles/manageiq-core.manageiq-vmdb/README.md]: @Fryguy - I can align the API version - how best to say that here?

Comment (gmcculloug) [reply content/ansible/roles/manageiq-core.manageiq-vmdb/README.md]: Maybe @abellotti can advise us here.

File: content/ansible/roles/manageiq-core.manageiq-vmdb/library/manageiq_vmdb.py (hunk 6 above)
Comment (syncrou) [reply content/ansible/roles/manageiq-core.manageiq-vmdb/library/manageiq_vmdb.py]: @Fryguy - If I remember correctly - we don't have control over what comes in via the playbook.  There are cases where a slug is used `href_slug::users/10`  or other cases where `https://manageiq_domain.local/api/users/10` could be used. I didn't want the url 'building' logic to muddle up what I thought should be a `url` property in the class.
I think at the time, I thought the complexity would be worse than it ended up, so the logic in this case isn't too much of a deal, but just giving the background.

File: content/ansible/roles/manageiq-core.manageiq-automate/meta/main.yml (hunk 3 above)
Comment (syncrou) [reply content/ansible/roles/manageiq-core.manageiq-automate/meta/main.yml]: Maybe I can pull the `galaxy_tags:` for this role - it obviously was copied and pasted from Galaxy.

**Comment by syncrou** [comment]
@miq-bot add_label garpindashvili/yes

**Comment by imtayadeway** [comment]
@gmcculloug I attempted to answer your question in the form of a PR: https://github.com/ManageIQ/manageiq-content/pull/277

**Comment by syncrou** [comment]
@Fryguy - Everything we discussed for this PR is taken care of.

**Comment by Fryguy** [comment]
The bot hasn't rerun on this repo, so it hasn't really used the yamllint stuff.  The only way to force it is to push a commit (just amend the commit and force push).

**Comment by syncrou** [comment]
@Fryguy - Made a couple of changes suggested by the new yamllint rules.  Everything is green now again.

**Comment by simaishi** [comment]
Gaprindashvili backport details:

```
$ git log -1
commit 3bc2d372216956eea3ff4f3b2584a8f3de4d31b8
Author: Jason Frey <fryguy9@gmail.com>
Date:   Thu Apr 12 11:28:35 2018 -0400

    Merge pull request #254 from syncrou/seed_ansible_playbooks
    
    Add in the miq. automate and vmdb roles
    (cherry picked from commit e045e8c31a30cee9c3f455da7f9221e7f30a504b)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1566658
```

**Comment by NickLaMuro** [comment]
Note:  Updated the PR title and description to match the changes that were made and mentioned in these comments:

- https://github.com/ManageIQ/manageiq-content/pull/254#issuecomment-379785145
- https://github.com/ManageIQ/manageiq-content/pull/254#issuecomment-379829026

**Pull request merged**